site-src/geps/gep-1282.md

```diff
@@ -0,0 +1,79 @@
+# GEP-696: GEP template
+
+* Issue: [#1282](https://github.com/kubernetes-sigs/gateway-api/issues/1282)
+* Status: Provisional
+
+(See definitions in [Kubernetes KEP][kep-status].
+
+[kep-status]: https://github.com/kubernetes/enhancements/blob/master/keps/NNNN-kep-template/kep.yaml#L9
+
+## TLDR
+
+End-user feedback of Gateway API has presented a trend: some users are interested in defining functionality that describes capabilities of the service more than the route you take to get to the service. In order to support these more advanced route functions, implementations need programmatic access to deeper insights about a Service’s (that is: a Kubernetes Service that would be used in an *Route backendRef) capabilities, such as TLS or websocket details. The purpose of this GEP is to add APIs for capturing those Service capabilities to enable more complex routing composition.
```

Comment:
> some users are interested in defining functionality that describes capabilities of the service more than the route you take to get to the service

To me this statement seems to demand more information about "why": it's not clear to me after reading the entire GEP so far _why_ users are interested in this functionality.

Reply:
It's a bit difficult to explain _why_ people want this things without talking in more detail about what they want, so I guess I'll need to fix this later in the GEP?

Bascially, the thing here is: Users have asked for stuff like backend TLS termination, and we don't have anywhere to put that in the current API. There are a few related features, so this GEP is about trying to make a broad solution to that class of problem - adding the description of extra capabilities to the backend.

I don't know how to say that more concisely than what I have here though.

Reply:
I've had a crack after updating the rest of the GEP, PTAL.